interfaces/builtin/dotfiles.go

```diff
@@ -0,0 +1,132 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/snap"
+)
+
+const dotfilesSummary = `allows access to specific hidden files in the home directory`
+
+const dotfilesBaseDeclarationSlots = `
+  dotfiles:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
```

Comment:
@jdstrand @mvo5 @niemeyer perhaps you can agree to this while physically together so this can move on. I agree with Jamie that we ought to be very careful here.

Reply:
The issue I think is that a user will not always be able to  make a good decision whether connecting files is appropriate or not, given that is very generic and abstract, and we don't have mechanisms atm to see/inform easily what attribute values (paths here) are involved afaik.

This is an area where prompting could help, OTOH here paths are hard-coded so it seems appropriate for the snap-declaration to control this.

So in summary I think it would be appropriate to start very carefully and actually block installation of snap using "files". Which would then make our task to provide appropriate snap-declarations and rules on a case-by-case basis.

@jdstrand any further thoughts on this?

Reply:
@pedrons - +1 for installation constraint. When @niemeyer and I initially discussed it, we definitely wanted to carefully guard this. It isn't meant to be a get out of jail free card (&lt;/rimshot&gt;) but something where snap declarations carefully control what additional accesses are needed *and* humans have a conversation about the paths and if they should be writable. If we start more restricted, we can always open it up later if we want; it is much harder going the other way.